.github/workflows/ci.yml

```diff
@@ -0,0 +1,755 @@
+        env:
+          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  java-app-encryption:
+    name: Build Java Application Encryption
+    needs: java-secure-memory
+    runs-on: ubuntu-latest
+    container:
+      image: adoptopenjdk/maven-openjdk11
+      options: --ulimit core=-1 --ulimit memlock=-1:-1
+    steps:
+    - name: Checkout the repository
+      uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+    - name: Cache Java packages
+      uses: actions/cache@1a9e2138d905efd099035b49d8b7a3888c653ca8
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Build
+      run: |
+        cd java/app-encryption
+        ./scripts/clean.sh
+        ./scripts/build.sh
+    - name: Unit tests
+      run: |
+        cd java/app-encryption
+        ./scripts/test.sh
+    - name: Integration tests
+      run: |
+        cd java/app-encryption
+        ./scripts/integration_test.sh
+
+  release-java-app-Encryption:
+    if: contains( github.ref, 'release' )
+    name: Release Java App Encryption
+    needs: java-app-encryption
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+      - name: Fetch all tags
+        run: git fetch --prune --unshallow --tags
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@8764a52df183aa0ccea74521dfd9d506ffc7a19a
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }} # Value of the GPG private key to import
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
+      - name: Publish Java AppEncryption
+        run: |
+          cd java/app-encryption
+          BASE_VERSION=$(mvn -q -DforceStdout help:evaluate -Dexpression=project.version)
+          VERSION_SUFFIX=`echo ${BASE_VERSION} | cut -f2 -d'-'`
+          BRANCH=`echo ${GITHUB_REF#refs/heads/}`
+          if [[ "${BRANCH}" =~ release-.* && "${VERSION_SUFFIX}" != "alpha" ]]; then
+            ./scripts/release_prod.sh
+          fi
+        env:
+          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  java-reference-app:
+    name: Java Reference Application
+    needs: java-app-encryption
+    runs-on: ubuntu-latest
+    container:
+      image: adoptopenjdk/maven-openjdk11
+      options: --ulimit core=-1 --ulimit memlock=-1:-1
```

Comment:
It occurs to me that we probably don't need to spin up a container for this step, since there are no tests being run in this job. IIRC the container and specifically the options listed here were only necessary when running the tests.

Reply:
This doesn't necessarily have to be changed before we merge, since this works for now. But if not then I think we should have a fast-follow for it.